hca/full_download.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+import sys
+
+from io import open
+
+
+def eprint(*args, **kwargs):
+    """Print to stderr."""
+    print(*args, file=sys.stderr, **kwargs)
+
+
+class FullDownload:
+    """Functions needed to fully add this functionality to the command line parser."""
+
+    CONSOLE_ARGUMENT = "download"
+    UUID = "uuid"
+
+    @classmethod
+    def add_parser(cls, subparsers):
+        """Call from parser.py to create the parser."""
+        subparser = subparsers.add_parser(
+            cls.CONSOLE_ARGUMENT,
+            help="Download a full bundle or file to local."
+        )
+
+        subparser.add_argument(
+            cls.UUID,
+            help="UUID of the folder or file you want to download."
+        )
+
+        subparser.add_argument(
+            "--name",
+            help="Name you're going to assign to the bundle/file you're downloading."
+        )
+
+        subparser.add_argument(
+            "--replica",
+            help="Which cloud to download from first. One of 'aws', 'gc', or 'azure'.",
+            default="aws"
+        )
+
+    @classmethod
+    def _download_files(cls, files, folder, api, replica):
+        """Use the API class to make a put-files request on each of these files."""
+        for file in files:
```

Comment:
Planning on changing to `file_`. Would it be better to change to a different name altogether?